stable/database/files/prepare-archive

```diff
@@ -0,0 +1,51 @@
+#!/bin/sh
+
+fail() {
+    echo "$1" 1>&2
+    exit 1
+}
+
+# Check whether data migration has already been performed
+db_dir="/var/opt/nuodb/archive/$NUODB_DOMAIN/$DB_NAME"
+if [ -f "$db_dir/migrated" ]; then
+    echo "No data migration to perform"
+    exit 0
+fi
+
+# Look for archive directories based on presence of info.json
+set -- $(find /var/opt/nuodb/archive -name info.json -exec dirname {} \;)
+if [ $# -eq 0 ]; then
+    echo "No archive to migrate"
+    mkdir -p "$db_dir"
+    touch "$db_dir/migrated"
+    exit 0
+fi
+[ $# -eq 1 ] || fail "Expected 1 archive, found $#: $*"
+source_archive="$1"
+
+# If source directory is different from target, move it
+if [ "$source_archive" != "$db_dir" ]; then
+    if [ -e "$db_dir" ] && ! rmdir "$db_dir"; then
+        fail "Unable to delete $db_dir"
+    fi
+    mv "$source_archive" "$db_dir"
```

Comment:
Should we account for an external journal for the source and the target database?